network/src/router.rs

```diff
@@ -289,13 +268,13 @@ impl<P: ProvideRuntimeApi + Send, E, N, T> TableRouter for Router<P, E, N, T> wh
 {
 	type Error = io::Error;
 	type FetchValidationProof = validation::PoVReceiver;
-	type FetchBlockChunk = validation::ChunkReceiver;
 
-	fn local_collation(&self, collation: Collation, outgoing: OutgoingMessages, chunks: (ValidatorIndex, &Vec<ErasureChunk>)) {
+	// We have fetched from a collator and here the receipt should have been already formed.
+	fn local_collation(&self, collation: Collation, receipt: CandidateReceipt, outgoing: OutgoingMessages, chunks: (ValidatorIndex, &Vec<ErasureChunk>)) {
 		// produce a signed statement
-		let hash = collation.receipt.hash();
+		let hash = receipt.hash();
 		let validated = Validated::collated_local(
-			collation.receipt,
+			receipt,
 			collation.pov.clone(),
```

Comment:
all we need from the collation is the pov-block. we shouldn't pass the whole collation, since it's a bit too cluttered

Reply:
that's what happens here, isn't it? the receipt and the pov-block are passed to `collated_local`